yarn-project/txe/src/oracle/txe_oracle.ts

```diff
@@ -239,10 +261,14 @@ export class TXE implements TypedOracle {
     sideEffectsCounter = this.sideEffectCounter,
     isStaticCall = false,
   ) {
-    const db = await this.#getTreesAt(blockNumber);
-    const previousBlockState = await this.#getTreesAt(blockNumber - 1);
+    if (blockNumber >= this.blockNumber) {
```

Comment:
Mmm, this is weird. Shouldn't we issue an error or something if someone tries to ask for stuff in the future?